File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/cards/CardWidget.java
package com.sup.dev.android.views.cards;

import android.content.Context;
import android.view.View;

import com.sup.dev.android.views.widgets.Widget;
import com.sup.dev.android.views.widgets.WidgetViewWrapper;

public class CardWidget extends Card implements WidgetViewWrapper{

    private final Widget widget;

    public CardWidget(Widget widget){
        this.widget=  widget;
    }

    @Override
    public int getLayout() {
        return 0;
    }

    @Override
    public View instanceView(Context context) {
        return widget.getView();
    }

    @Override
    public void bindView(View view) {
        widget.onShow();
    }

    @Override
    public <K extends WidgetViewWrapper> K hideWidget() {
        adapter.remove(this);
        return (K) this;
    }

    @Override
    public <K extends WidgetViewWrapper> K setWidgetCancelable(boolean cancelable) {
        return (K) this;
    }

    @Override
    public <K extends WidgetViewWrapper> K setWidgetEnabled(boolean enabled) {
        return (K) this;
    }
}

File: src/main/java/com/sup/dev/android/views/cards/CardWidget.kt
package com.sup.dev.android.views.cards

import android.view.View
import android.view.ViewGroup
import com.sup.dev.android.views.widgets.Widget
import com.sup.dev.android.views.widgets.WidgetViewWrapper

class CardWidget(private val widget: Widget) : Card(0), WidgetViewWrapper<CardWidget> {

    override fun instanceView(vParent: ViewGroup): View {
        return widget.view
    }

    override fun bindView(view: View) {
        super.bindView(view)
        widget.onShow()
    }

    override fun hide(): CardWidget {
        adapter!!.remove(this)
        return this
    }

    override fun setWidgetCancelable(cancelable: Boolean): CardWidget {
        return this
    }

    override fun setWidgetEnabled(enabled: Boolean): CardWidget {
        return this
    }
}
